### PR TITLE
Fix GET step run endpoint to return unhydrated response if requested

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR fixes a few hydration issues:
- When requesting an unhydrated step run in the `GET`, it always included the metadata
- When requesting a hydrated pipeline run, it always included the metadata for all steps (two layers of hydration)
- Some unnecessary server-side hydration calls

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by stefannica — APPROVED** [review]

File: src/zenml/zen_server/routers/steps_endpoints.py [hunk 1]
```diff
@@ -142,10 +142,16 @@ def get_step(
     Returns:
         The step.
     """
-    step = zen_store().get_run_step(step_id, hydrate=hydrate)
+    # We always fetch the step hydrated because we need the pipeline_run_id
+    # for the permission checks. If the user requested an unhydrated response,
+    # we later remove the metadata
+    step = zen_store().get_run_step(step_id, hydrate=True)
     pipeline_run = zen_store().get_run(step.pipeline_run_id)
```
Comment (bcdurak) [inline src/zenml/zen_server/routers/steps_endpoints.py:149]: I know it is in the metadata, but ideally, shouldn't the `pipeline_run_id` already be in the body of the response?

File: src/zenml/zen_stores/schemas/pipeline_run_schemas.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def to_model(
         )
 
         if self.deployment is not None:
-            deployment = self.deployment.to_model()
+            deployment = self.deployment.to_model(include_metadata=True)
```
Comment (bcdurak) [inline src/zenml/zen_stores/schemas/pipeline_run_schemas.py:301]: Why is this needed?

**Review by bcdurak — APPROVED** [review]
I'll approve to unblock the whole process, but put some questions there. Feel free to merge first, answer later, as you see fit.

File: src/zenml/zen_server/routers/steps_endpoints.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/zen_server/routers/steps_endpoints.py:149]: Ideally it would be, but then this might be breaking some stuff in the frontend which I don't want to risk right now

File: src/zenml/zen_stores/schemas/pipeline_run_schemas.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/zen_stores/schemas/pipeline_run_schemas.py:301]: It just saves a hydration call that would happen in the next line when we access some attributes of the deployment

Comment (bcdurak) [reply src/zenml/zen_stores/schemas/pipeline_run_schemas.py:301]: I see, thanks.
